game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProSortMoveOptionsUtils.java

```diff
@@ -125,195 +115,126 @@ private ProSortMoveOptionsUtils() {}
-          if (o1.getKey().getType().equals(o2.getKey().getType())) {
-            final boolean isAirUnit = UnitAttachment.get(o1.getKey().getType()).getIsAir();
-            if (isAirUnit) {
-              int distance1 = 0;
-              for (final Territory t : o1.getValue()) {
-                if (!attackMap.get(t).isCurrentlyWins()) {
-                  distance1 +=
-                      data.getMap()
-                          .getDistanceIgnoreEndForCondition(
-                              unitTerritoryMap.get(o1.getKey()),
-                              t,
-                              ProMatches.territoryCanMoveAirUnitsAndNoAa(player, data, true));
-                }
-              }
-              int distance2 = 0;
-              for (final Territory t : o2.getValue()) {
-                if (!attackMap.get(t).isCurrentlyWins()) {
-                  distance2 +=
-                      data.getMap()
-                          .getDistanceIgnoreEndForCondition(
-                              unitTerritoryMap.get(o2.getKey()),
-                              t,
-                              ProMatches.territoryCanMoveAirUnitsAndNoAa(player, data, true));
-                }
-              }
-              if (distance1 != distance2) {
-                return distance1 - distance2;
-              }
+          final UnitType unitType1 = unit1.getType();
+          final UnitType unitType2 = unit2.getType();
+
+          // If unit types are equal and are air, then sort by average distance.
+          if (unitType1.equals(unitType2) && UnitAttachment.get(unitType1).getIsAir()) {
+            final Predicate<Territory> predicate =
+                ProMatches.territoryCanMoveAirUnitsAndNoAa(player, data, true);
+            final Territory territory1 = unitTerritoryMap.get(unit1);
+            final Territory territory2 = unitTerritoryMap.get(unit2);
+            int distance1 = 0;
+            for (final Territory t : territories1) {
+              distance1 += data.getMap().getDistanceIgnoreEndForCondition(territory1, t, predicate);
+            }
+            int distance2 = 0;
+            for (final Territory t : territories2) {
+              distance2 += data.getMap().getDistanceIgnoreEndForCondition(territory2, t, predicate);
+            }
+            if (distance1 != distance2) {
+              return distance1 - distance2;
             }
           }
-          return o1.getKey().getType().getName().compareTo(o2.getKey().getType().getName());
+
+          return unitType1.getName().compareTo(unitType2.getName());
         });
     final Map<Unit, Set<Territory>> sortedUnitAttackOptions = new LinkedHashMap<>();
     for (final Map.Entry<Unit, Set<Territory>> entry : list) {
       sortedUnitAttackOptions.put(entry.getKey(), entry.getValue());
     }
     return sortedUnitAttackOptions;
   }
+
+  private static Collection<Territory> filterOutWinningTerritories(
```

Comment:
Any thoughts if `filterNonWinningTerritories` is a better name for this method?

Reply:
I actually prefer my version. Your version sounds a bit ambiguous to me (i.e. my first read of it is to mean that the input is non-winning territories and they're being filtered by some other unknown criteria, as opposed to the input being all territories and non-winning are kept). I think "filter out" wording in the current name doesn't have this ambiguity, so keeping it.

Reply:
How about `removeWinningTerritories`?

`FilterOut` could mean to "filter out" the winning territories from the rest and return those.

Or it could mean "filter out" as-in remove them.

It could be slightly confusing as to which one is meant.

Reply:
`removeWinningTerritories` works for me. Done!